common/src/main/java/whocraft/tardis_refined/common/tardis/manager/TardisPilotingManager.java

```diff
@@ -547,7 +551,7 @@ private void endFlightEarly(boolean dramatic) {
         float percentageY = (targetPosition.getY() - startingPosition.getY()) * percentage;
         float percentageZ = (targetPosition.getZ() - startingPosition.getZ()) * percentage;
 
-        TardisNavLocation newLocation = new TardisNavLocation(new BlockPos((int) percentageX,(int) percentageY, (int)percentageZ), this.targetLocation.getDirection(), percentage > 0.49f ? this.targetLocation.getLevel() : this.operator.getExteriorManager().getLastKnownLocation().getLevel() );
+        TardisNavLocation newLocation = new TardisNavLocation(new BlockPos((int) percentageX, (int) percentageY, (int) percentageZ), this.targetLocation.getDirection(), percentage > 0.49f ? this.targetLocation.getLevel() : this.operator.getExteriorManager().getLastKnownLocation().getLevel());
```

Comment:
Are the int casts needed? Could potentially make ``percentageY`` and ``percentageZ`` a double?

Reply:
BlockPos only uses int.

Reply:
I’m sure it has a doubles constructor for when you use 0.5 etc in commands

Reply:
Those are recorded as positions, which are Vector3s. 
![image](https://github.com/Jeryn99/TardisRefined/assets/34910888/d9ae4ca3-36af-47a0-8fbb-cff50e0b6013)